### PR TITLE
Add postAttesterSlashingV2

### LINKED ISSUE #8425: Add the new `PostAttesterSlashing` V2 API
Add new `PostAttesterSlashing` V2 API to handle new `IndexedAttestationElectra`
[Spec PR](https://github.com/ethereum/beacon-APIs/pull/448)

### PR DESCRIPTION
**Pull request by gfukushima**
## PR Description
This PR adds the post method for postAttesterSlashingV2 to the beacon API 

## Fixed Issue(s)

Fixes #8425

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v2/beacon/PostAttesterSlashingV2IntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      new DataStructureUtil(TestSpecFactory.createMinimalElectra());
+
+  @BeforeEach
+  public void setup() {
+    startRestAPIAtGenesis();
+  }
+
+  @Test
+  public void shouldReturnBadRequestWhenRequestBodyIsEmpty() throws Exception {
+    Response response = post(PostAttesterSlashingV2.ROUTE, jsonProvider.objectToJSON(""));
```
Comment (rolfyone) [inline data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v2/beacon/PostAttesterSlashingV2IntegrationTest.java]: final :)
i wont comment every one, but general through whole class :)

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/JsonTypeDefinitionBeaconRestApi.java [hunk 2]
```diff
@@ -242,6 +243,7 @@ private static RestApi create(
             .endpoint(new GetAttesterSlashings(dataProvider, spec))
             .endpoint(new GetAttesterSlashingsV2(dataProvider, schemaCache))
             .endpoint(new PostAttesterSlashing(dataProvider, spec))
+            .endpoint(new PostAttesterSlashingV2(dataProvider, schemaCache))
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/JsonTypeDefinitionBeaconRestApi.java]: we could even try deprecating here, i think potentially if hte new one is first, it should come up before the one we're killing off..

**Review by rolfyone — APPROVED** [review]
LGTM just nits

File: data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v2/beacon/PostAttesterSlashingV2IntegrationTest.java (hunk 1 above)
Comment (gfukushima) [reply data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v2/beacon/PostAttesterSlashingV2IntegrationTest.java]: thanks!

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/JsonTypeDefinitionBeaconRestApi.java (hunk 2 above)
Comment (gfukushima) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/JsonTypeDefinitionBeaconRestApi.java]: deprecated the V1. Changing the order here didn't make any difference on swagger at least

**Pull request merged**